internal/pkg/utils/http.go

```diff
@@ -139,17 +140,30 @@ func ParseTimeRangeOffsetLimit(c echo.Context, minOffset int, maxOffset int, min
 	if end < start {
 		return start, end, offset, limit, errors.NewCommonEdgeX(errors.KindContractInvalid, fmt.Sprintf("end's value %v is not allowed to be greater than start's value %v", end, start), nil)
 	}
-	offset, edgexErr = ParseQueryStringToInt(c, common.Offset, common.DefaultOffset, minOffset, maxOffset)
+	offset, limit, _, edgexErr = ParseGetAllObjectsRequestQueryString(c, minOffset, maxOffset, minLimit, maxLimit)
 	if edgexErr != nil {
 		return start, end, offset, limit, edgexErr
 	}
-	limit, edgexErr = ParseQueryStringToInt(c, common.Limit, common.DefaultLimit, minLimit, maxLimit)
+
+	return start, end, offset, limit, nil
+}
+
+func ParseQueryStringTimeRangeOffsetLimit(c echo.Context, minOffset int, maxOffset int, minLimit int, maxLimit int) (start int, end int, offset int, limit int, edgexErr errors.EdgeX) {
+	// TODO: Create min and max constants for start and end in go-mod-core-contracts?
+	start, edgexErr = ParseQueryStringToInt(c, common.Start, 0, 0, math.MaxInt64)
```

Comment:
since the function signature of ParseQueryStringToInt only accepts `int`, passing `math.MaxInt64` to the function will cause issue on arm32 machine.
On arm32 machine, int range equals to int32

Reply:
use ParsePathParamToInt func

Reply:
I opened this issue https://github.com/edgexfoundry/edgex-go/issues/4845
we could handle the int and int64 in the later PR

Reply:
Since start, end, offset, and limit here are all URL parameters, we should use the `ParseQueryStringToInt` func, e.g. `/api/v3/job/all?start=1722479643&end=1722479650`.

Maybe we should make the design of this API consistent with the design of the reading and event APIs?
e.g. `/api/v3/event/start/:start/end/:end`

Reply:
let's keep the same API, and handle the int64 in https://github.com/edgexfoundry/edgex-go/issues/4845